oarepo_oidc_einfra/cli.py

```diff
@@ -0,0 +1,214 @@
+        }
+        _datastore.create_user(**kwargs)
+        db.session.commit()  # type: ignore
+
+        user = User.query.filter_by(email=email).one()
+
+    identity = UserIdentity.query.filter_by(
+        method="e-infra", id=einfra_id, id_user=user.id
+    ).first()
+    if not identity:
+        UserIdentity.create(
+            user=user,
+            method="e-infra",
+            external_id=einfra_id,
+        )
+        db.session.commit()  # type: ignore
+
+
+@einfra.command("import_dump_users")
+@click.argument("dump_path")
+@with_appcontext
+def import_dump_users(dump_path: str) -> None:
+    """Import users from a dump file.
+
+    :param dump_path: Path to the dump file in the S3 bucket.
+
+    Note: this cli command is usually not used in the application, it is here for testing purposes.
+    """
+    client = current_einfra_oidc.dump_boto3_client
+
+    with BytesIO() as obj:
+        client.download_fileobj(
+            Bucket=current_einfra_oidc.dump_s3_bucket,
+            Key=dump_path,
+            Fileobj=obj,
+        )
+        obj.seek(0)
+        data = json.loads(obj.getvalue().decode("utf-8"))
+
+    for user_data in data["users"].values():
+        einfra_id = user_data["attributes"].get(
+            "urn:perun:user:attribute-def:virt:login-namespace:einfraid-persistent"
+        )
+        email = user_data["attributes"].get(
+            "urn:perun:user:attribute-def:def:preferredMail"
+        )
+        if not email or not einfra_id:
+            continue
+        print("Importing user", email, einfra_id)
+        _add_einfra_user(email, einfra_id)
+
+
+@einfra.command("synchronize_community")
+@click.argument("community_slug")
+@with_appcontext
+def synchronize_community(community_slug: str) -> None:
+    """Re-synchronize a community to Perun."""
+    community = current_communities.service.read(system_identity, community_slug)
+    synchronize_community_to_perun(str(community.id))
+
+
+@einfra.command("synchronize_all_communities")
+@with_appcontext
+def synchronize_all_communities() -> None:
+    """Re-synchronize all communities to Perun."""
+    from tqdm import tqdm
+
+    community_list = CommunityMetadata.query.all()
+    for community in tqdm(community_list):
+        synchronize_community_to_perun(str(community.id))
+
+
+@einfra.command("resend_invitation")
+@click.argument("community_slug")
+@click.argument("email")
+@with_appcontext
+def resend_invitation(community_slug: str, email: str) -> None:
+    """Resend an invitation to a user to a community.
+
+    :param community_slug: Slug of the community.
+    :param email: Email of the user.
+    """
+    community = CommunityMetadata.query.filter_by(slug=community_slug).one()
+    user = User.query.filter_by(email=email).one()
+    member = MemberModel.query.filter_by(
+        user_id=user.id, community_id=community.id
+    ).one()
+    request_id = member.request_id
+    create_aai_invitation(str(request_id))
```

Comment:
Wouldn't create_aai_invitation create duplicit invitation for the member? Shouldn't we just fetch invitation request and re-send notification from it?